### PR TITLE
replace with pathogen adjusted version of traceability vocab + some a…

### PR DESCRIPTION
**Pull request by Therecanbeonlyone1969**
…dditions to automate further and incorporate evidence docs

### CODE REVIEW

**Comment by Therecanbeonlyone1969** [comment]
@tplooker @cstoecker this is a full redo based on the traceability vocab but adjusted for pathogen + added the following:

- ability to include evidence documents (still based on a static schema, will make this dynamic later on)
- schemas are built on sub-schemas -> kept a few subschemas from the traceability vocab that are general and are useful for more complex schemas
- this has the CI from traceability as well as includes credential verification and json-ld checker etc.
- It does not yet include bbs+ signatures but I will add that tomorrow or early next week. Just did not have time yet but I know how to do it.

**Review by cstoecker — APPROVED** [review]
@tplooker 

- This is our proposal of how to blend schemata of the CCG traceability vocab with pathogen vocab
- test pass types can be PCR, antigen, antibody or other medical test
- it is pretty flexible re/ use of evidence documents for offloading data of the VC

We expect the need to distinguish between 

1. **medical test documentation credential**, i.e. state of a medical condition of a test person, this is what is part of the test passes and 
2. an **access policy credential**, i.e. based on the medical test documentation and policies or regulatory rules where is a person allowed to enter. This type of credential will be important considering all the COVID edge cases such as mutations, test sensitivities and medicinal effects.

This distinction is not part of the current political debate. We might need to introduce this distinction at a later point in time.

**Review by cstoecker — APPROVED** [review]
Approved to merge with main ... fixing the CI pipeline with linter to be done as a next step.

added **CovidTestRegristationCredential** will be used for processing edge cases in case an official test product or test type registration will be revoked by the health regulatory body (e.g. BfArM [Antigen Registry](https://antigentest.bfarm.de/ords/f?p=101:100:4195387605300:::::&tz=2:00) in Germany).

File: bbs+.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        secureRandom: () => {
+            return Buffer.from(ISSUER_KEY_PAIR_SEED_BASE64,'base64');
+        },
+    });
+    keypair.id = keypair.controller + keypair.id;
+    const suite = new BbsBlsSignature2020({
+        key: keypair,
+    });
+
+    console.log (keypair);
```
Comment (tplooker) [inline bbs+.js:18]: Unsure what this function is doing? Just logging the generated key pair?

**Comment by tplooker** [comment]
> pathogen access policy credential, i.e. based on the medical test documentation and policies or regulatory rules: "Where, how and under which constraints is a person allowed to enter a facility?" This type of credential will be important considering all the COVID edge cases as well as mutations, test sensitivities and medicinal effects.

We should discuss this some more, I feel like this is leaking the abstraction that I would have thought a COVID test result credential should be constrained to? For example IMO access control decisions I would have imagined would be determined by the verifier rather than imposed upon by the verifier?

In any case I think we should merge the PR as it is quite large in its current form.

File: bbs+.js (hunk 1 above)
Comment (Therecanbeonlyone1969) [reply bbs+.js]: @tplooker ... forgot to remove this file. was testing out stuff on bbs+ and forgot to remove after I was done

**Comment by Therecanbeonlyone1969** [comment]
> > pathogen access policy credential, i.e. based on the medical test documentation and policies or regulatory rules: "Where, how and under which constraints is a person allowed to enter a facility?" This type of credential will be important considering all the COVID edge cases as well as mutations, test sensitivities and medicinal effects.
> 
> We should discuss this some more, I feel like this is leaking the abstraction that I would have thought a COVID test result credential should be constrained to? For example IMO access control decisions I would have imagined would be determined by the verifier rather than imposed upon by the verifier?
> 
> In any case I think we should merge the PR as it is quite large in its current form.

@tplooker ... everything is an initial draft ... let's discuss on a call. Get this merged and then we have a solid foundation to expand.

**Comment by cstoecker** [comment]
Initial merge:

A pathogen is any organism that can produce disease. A pathogen may also be referred to as an **infectious agent**, or simply a germ or virus. The **pathogen vocabulary** shall cover pathogen related credentials: 
- medical vaccination event
- medical test & immunization measurement event (PCR, antibody, antigen, lateral flow, other test)
- infection risk and access control 

The initial merge covers the following artifacts for "medical test & immunization measurement event":
- CovidTestPassStandardCredentialDetailed schema
- CovidTestPassStandardCredentialMinimal schema
- CovidTestPassEvidenceDocument schema; for off-loading more detailed medical event information (e.g. test center, test details)
- CovidTestRegistrationCredential schema
- Related schemata such as organisation, location, product, legal entity

The pathogen vocab is inspired building upon the following community work
- [Vaccination Certificate Vocabulary](https://w3c-ccg.github.io/vaccination-vocab/)
- [WHO Smart Vaccination Card Work](]https://www.who.int/publications/m/item/interim-guidance-for-developing-a-smart-vaccination-certificate) 
- [CCG Traceability Vocabulary](https://github.com/w3c-ccg/traceability-vocab)
- [Interoperability of health certificates trust framework](https://ec.europa.eu/health/sites/health/files/ehealth/docs/trust-framework_interoperability_certificates_en.pdf) of EU E-Health Network
